common/prometheus-server/templates/_helpers.tpl

```diff
@@ -1,32 +1,61 @@
-{{/* Name of the Prometheus. */}}
+{/* Name of the Prometheus. */}}
 {{- define "prometheus.name" -}}
-{{- required ".Values.name missing" .Values.name -}}
+{{- $name := index . 0 -}}
+{{- $root := index . 1 -}}
+{{- if and (not $root.Values.names) (not $root.Values.global.targets) (not $root.Values.name) -}}
+{{- fail "Connot create any Prometheus resource. Please define a name or at least one list element to names or global.targets" -}}
+{{- end -}}
+{{/* vmware prometheis need additional renaming */}}
+{{- if $root.Values.vmware -}}
+{{- $vropshostname := split "." $name -}}
+vmware-{{ $vropshostname._0 | trimPrefix "vrops-" }}
+{{- else -}}
+{{- $name -}}
+{{- end -}}
 {{- end -}}
 
 {{/* Fullname of the Prometheus. */}}
 {{- define "prometheus.fullName" -}}
+{{- $name := index . 0 -}}
+{{- $root := index . 1 -}}
+{{/* vmware prometheis need additional renaming */}}
+{{- if $root.Values.vmware -}}
+{{- $vropshostname := split "." $name -}}
+prometheus-vmware-{{ $vropshostname._0 | trimPrefix "vrops-" }}
+{{- else -}}
 prometheus-{{- (include "prometheus.name" .) -}}
 {{- end -}}
+{{- end -}}
 
 {{/* External URL of this Prometheus. */}}
 {{- define "prometheus.externalURL" -}}
-{{- if and .Values.ingress.hosts .Values.ingress.hostsFQDN -}}
+{{- $name := index . 0 -}}
+{{- $root := index . 1 -}}
+{{- if and $root.Values.ingress.hosts $root.Values.ingress.hostsFQDN -}}
 {{- fail ".Values.ingress.hosts and .Values.ingress.hostsFQDN are mutually exclusive." -}}
 {{- end -}}
-{{- if .Values.ingress.hosts -}}
-{{- $firstHost := first .Values.ingress.hosts -}}
-{{- required ".Values.ingress.hosts must have at least one hostname set" $firstHost -}}.{{- required ".Values.global.region missing" .Values.global.region -}}.{{- required ".Values.global.domain missing" .Values.global.domain -}}
+prometheus-{{- $name -}}.{{- required "$root.Values.global.region missing" $root.Values.global.region -}}.{{- required "$root.Values.global.domain missing" $root.Values.global.domain -}}
```

Comment:
`.Values.ingress.hostsFQDN` or `.Values.ingress.hosts` are not being taken into consideration here. We're checking for them L34, but the names are independently set in L37.

I've read the comment in [values](https://github.com/sapcc/helm-charts/blob/0a359337be238b9b70ed5e89abe43152f0fc3b58/common/prometheus-server/values.yaml#L87) and it is a smart move, to technically disable this functionality if multiple prometheis are deployed, but L37 should use one of `.Values.ingress.hostsFQDN` or `.Values.ingress.hosts` in the default cause (1 name). If it's more, then we go like implemented. Or am I missing something here?

Reply:
You're right, the first host in `.Values.ingress.hosts` must be considered. `.Values.ingress.hostsFQDN` does not need an externalURL helper because the FQDN is already specified.